integration/tests/basic/evm/test_solana_interoperability.py

```diff
@@ -303,3 +412,251 @@ def test_limit_of_simple_instr_in_one_trx(self, call_solana_caller, counter_reso
+                AccountMeta(Pubkey(counter_resource_address), is_signer=False, is_writable=True),
+            ],
+            data=bytes([0x1]),
+        )
+        serialized = serialize_instruction(COUNTER_ID, instruction)
+
+        tx = self.web3_client.make_raw_tx(sender.address)
+        instruction_tx = call_solana_caller.functions.executeAndDoSomeIterativeActions(iterations,
+                                                                                       lamports,
+                                                                                       serialized).build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender, instruction_tx)
+        assert resp["status"] == 1
+        event_logs = call_solana_caller.events.LogBytes().process_receipt(resp)
+        assert int.from_bytes(event_logs[0].args.value, byteorder="little") == next(get_counter_value)
+
+    def test_iterative_actions_after_solana_call_exceed_accounts_limit(self, counter_resource_address, call_solana_caller):
+        iterations = 53
+        sender = self.accounts[0]
+        lamports = 0
+
+        instruction = Instruction(
+            program_id=COUNTER_ID,
+            accounts=[
+                AccountMeta(Pubkey(counter_resource_address), is_signer=False, is_writable=True),
+            ],
+            data=bytes([0x1]),
+        )
+        serialized = serialize_instruction(COUNTER_ID, instruction)
+
+        tx = self.web3_client.make_raw_tx(sender.address)
+
+        with pytest.raises(
+            web3.exceptions.ContractLogicError,
+            match="too many accounts: 65 > 64",
+        ):
+            call_solana_caller.functions.executeAndDoSomeIterativeActions(iterations, lamports, serialized).build_transaction(tx)
+
+    def test_iterative_actions_and_multiple_solana_calls(self, counter_resource_address, call_solana_caller):
+        solana_calls = 5
+        iterations = 20
+        sender = self.accounts[0]
+        lamports = 0
+        instruction = Instruction(
+            program_id=COUNTER_ID,
+            accounts=[
+                AccountMeta(Pubkey(counter_resource_address), is_signer=False, is_writable=True),
+            ],
+            data=bytes([0x1]),
+        )
+        serialized = serialize_instruction(COUNTER_ID, instruction)
+
+        tx = self.web3_client.make_raw_tx(sender.address)
+        instruction_tx = call_solana_caller.functions.executeMultipleInIterativeMode(solana_calls,
+                                                                                     iterations,
+                                                                                     lamports,
+                                                                                     serialized).build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender, instruction_tx)
+        assert resp["status"] == 1
+        assert len(call_solana_caller.events.LogBytes().process_receipt(resp)) == solana_calls
```

Comment:
does it work when you run all tests?
you use counter_resource_address for several tests and it means  len(call_solana_caller.events.LogBytes().process_receipt(resp)) == solana_calls + N

N - previous runs of COUNTER_ID program